features/storage/TESTS/kvstore/direct_access_devicekey_test/main.cpp

```diff
@@ -198,6 +198,13 @@ void test_direct_access_to_devicekey_tdb_last_two_sectors()
     uint32_t flash_bd_size = flash_bd_end_address - flash_bd_start_address;
 
     FlashIAPBlockDevice *flash_bd = new FlashIAPBlockDevice((bd_addr_t)flash_bd_start_address, (bd_size_t)flash_bd_size);
+    flash_bd->init();
+    uint32_t erase_prog_ratio = flash_bd->get_erase_size(flash_bd->size() - 1) / flash_bd->get_program_size();
```

Comment:
This calculates and checks only the last sector erase_prog_ratio but on a  very rare occasion, the second sector may have different erase_prog_ratio due to different sector size. I recommend adding a calculation and checkup also for the second sector.

Reply:
Good catch, fixed now. Please review.